spot-ml/src/main/scala/org/apache/spot/netflow/model/FlowSuspiciousConnectsModel.scala

```diff
@@ -43,19 +46,37 @@ class FlowSuspiciousConnectsModel(topicCount: Int,
                                   ibytCuts: Array[Double],
                                   ipktCuts: Array[Double]) {
 
+
   def score(sc: SparkContext, sqlContext: SQLContext, inDF: DataFrame): DataFrame = {
 
 
-    val ipToTopicMixBC = sc.broadcast(ipToTopicMix)
+    import sqlContext.implicits._
+    val ipToTopicMixRDD: RDD[(String, Array[Double])] = sc.parallelize(ipToTopicMix.toSeq)
+    val ipToTopicMixDF = ipToTopicMixRDD.map({ case (doc, probabilities) => IpTopicMix(doc, probabilities) }).toDF
+
+
     val wordToPerTopicProbBC = sc.broadcast(wordToPerTopicProb)
 
 
-    val scoreFunction =
-      new FlowScoreFunction(timeCuts,
+    val dataWithSrcTopicMix = {
```

Comment:
dataWithSrcTopicMix should be dataWithTopicMix or something like that since it contains both source and destination IP topic mixes or am I missing something?